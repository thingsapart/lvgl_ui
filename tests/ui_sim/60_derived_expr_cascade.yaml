# TICKS: 2
- type: data-binding
  state:
    - a: 1.0
    - b: { derived_expr: [mul, a, 2] }
    - c: { derived_expr: [add, b, 1] }
  actions:
    - inc_a:
        inc: { a: 1.0 }
  schedule:
    - { tick: 1, action: inc_a }
    - { tick: 2, action: inc_a }
