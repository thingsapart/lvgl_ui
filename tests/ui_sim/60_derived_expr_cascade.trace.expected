--- UI-Sim Trace Start ---
STATE_SET: a = 1.000 (old: null)
STATE_SET: b = 2.000 (old: null)
STATE_SET: c = 3.000 (old: null)
NOTIFY: a = 1.000
NOTIFY: b = 2.000
NOTIFY: c = 3.000

--- TICK 1 ---
ACTION: inc_a value=null
STATE_SET: a = 2.000 (old: 1.000)
STATE_SET: b = 4.000 (old: 2.000)
STATE_SET: c = 5.000 (old: 3.000)
NOTIFY: a = 2.000
NOTIFY: b = 4.000
NOTIFY: c = 5.000

--- TICK 2 ---
ACTION: inc_a value=null
STATE_SET: a = 3.000 (old: 2.000)
STATE_SET: b = 6.000 (old: 4.000)
STATE_SET: c = 7.000 (old: 5.000)
NOTIFY: a = 3.000
NOTIFY: b = 6.000
NOTIFY: c = 7.000

--- UI-Sim Trace End ---
